pkg/plugin/test-keeper/plugin/file_pattern_test.go

```diff
@@ -0,0 +1,76 @@
+package plugin_test
+
+import (
+	"fmt"
+
+	"github.com/arquillian/ike-prow-plugins/pkg/plugin/test-keeper/plugin"
+	. "github.com/onsi/ginkgo"
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("File pattern features", func() {
+
+	var it = func(filePattern, expectedRegex string) {
+		parsed := plugin.ParseFilePatterns([]string{filePattern})
+		Expect(parsed).To(ConsistOf(plugin.FilePattern{Regex: expectedRegex}))
+	}
+
+	Context("File pattern parsing", func() {
+
+		table.DescribeTable(
```

Comment:
Can we also think about covering `func (matcher *FilePattern) Matches(filename string) bool` the same way?

Reply:
I was thinking about it. I had also an idea to somehow incorporate it into the very first assertion - something like:
```go
it,
should().
	parsePattern("**/*Test.java").
	toRegex(`.*/[^/]*Test\.java$`, 
		matching(
			"src/test/java/org/MyTest.java",
			"!src/test/java/org/AnyClass.java"),
```
but 

* I didn't want to mix two things in one test.
* I realized that the logic of `Matches` method itself is verified in `matchers_test.go` anyway
* I had problems with writing the test that would bring an additional quality but didn't introduce too many code duplication and a lot work (I know I'm lazy)

so I decided to change a few file patterns in several other tests, so the logic would be verified.

Reply:
Yes, we cover it level higher, so that's good. Maybe we can review existing tests and see if we are not missing some important cases.